### DISCUSSION TITLE
Feature suggestion: automatically apply delay for overrunning events

### DISCUSSION THREAD

**Discussion by markspolakovs** [cpvalente/ontime#841] [Discussion]
I'd be happy to write a PR to implement this - opening this issue to discuss the idea.

In our workflow (@ystv) we often have segments (represented by ontime events) that overrun, and then we simply move on to the next event once the presenter concludes. It'd be useful if ontime were to automatically adjust the start/end times of all following events when we advance from an event that's overrunning.

I'd imagine this would be implemented by adding a delay block automatically after clicking "Next" on an overrunning event (assuming the next action is "start next"), with the delay amount being the amount the event overran by. Open to feedback before I start implementing though.

**Comment by markspolakovs** [comment]
Only realising after submitting that this may be a duplicate of #100, or an alternate implementation of the same idea... 🤦

**Comment by cpvalente** [comment]
Hi @markspolakovs , thank you for this.

I believe we had some other features that considered a similar approach to this.
As you correctly mention, some of those thoughts are in #100 

We have tried to improve this workflow in v3 with a few additions.
**Offset data** set gives feedback to the user of how far light or heavy our rundown is
**Link** feature allows having the rundown being more dynamic, where we can define dependencies in schedule (ie, an event starts at a specific time, or whenever the previous finishes)

There will still be room to improve, but I think, at this point, we should consider v3 as our baseline.
I would suggest that maybe you can checkout v3 and give us some feedback? Your PR effort would definitely be appreciated and likely more productive then
